### PR TITLE
PAPILIO-91-task

### PR DESCRIPTION
**Pull request by p-carrier**
## General

Some of the website components and pages have been created before the backend endpoints associated with them were created. This is creating some discrepancies between what we were expecting and what is already created. To make sure that the backend is called at the right position, we should verify all the endpoints and implement them if they are missing. It would also be a good time to refactor some of the component to reduce duplication.

_Issue number_: Papilio-91

## Task description
- verify getEmployee, getActivity backend connection
- improve test coverage for easier refactoring
- add interaction events on delete employee
- Improve UX on section change in employeeDashboard
- Add automated acceptance test
- Genereal code refactoring

## Manual Test

To test you need to have an employee account with an Admin role. If you don't have one, you can ask me to provide one.
Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)

1. start the backend
`npm run build && npm start`
2. start the website
`npm start`
3. On the main page navigate to the login page
4. Insert the admin account information
5. Once logged in, navigate to the Employee page
6. Verify that you have some employees in your table
7. Click on the action button (add employee, delete employee), they should change to close once clicked
8. Navigate to Activity manager
9. Verify that some activities are in the table
10. Click the add activity button, it should change to close once clicked

## Testing
To run all the unit tests, run the following code: `npm test`
To make sure that the tests have reach the appropriate level of coverage, run the following code: `npm run coverage`
To test the automated acceptance test: (make sure to `npm install` to have cypress installed on your machine)
1. start the server `npm start`
2. start the application `npm start`
3. run `npx cypress run`

### CODE REVIEW

**Comment by mth-1012** [comment]
Seems to be behaving correctly following the steps you provided. So good job!

One thing though.

image

The end time could also use a conversion to readable date/time. I also think time is as important as date so don't eliminate it.

File: src/api/apiLayer.tsx [hunk 1]
```diff
@@ -40,6 +40,16 @@ export async function getActivites(
     method: 'GET',
   })
     .then(async (res) => await res.json())
+    .then(({ activities }) =>
+      activities.map((activity: any) => ({
+        id: activity.id?.toString(),
+        title: activity.title,
+        startTime: formatDate(activity.startTime),
+        endTime: activity.endTime ?? '',
```
Comment (mth-1012) [inline src/api/apiLayer.tsx:48]: bro why did you do 1 but not both 🥲

**Comment by mth-1012** [comment]
Another problem I found with the UI.

Before:

image

After the change from this PR:

image

I know, very nit-picking, and I knew that you made some changes with the layout in the code. I just gotta say it... There's a new horizontal scroll appearing.

It happens to every screen size, and every zoom level (25-200%)

File: src/api/apiLayer.tsx (hunk 1 above)
Comment (p-carrier) [reply src/api/apiLayer.tsx:48]: I looked in the /addActivity and endTime was not there. I thought it was deleted so I did not do anything for it and forgot to ask why this was the case. I'll add it back 😦

**Comment by p-carrier** [comment]
> Another problem I found with the UI.
> 
> Before:
> 
> image
> 
> After the change from this PR:
> 
> image
> 
> I know, very nit-picking, and I knew that you made some changes with the layout in the code. I just gotta say it... There's a new horizontal scroll appearing.
> 
> It happens to every screen size, and every zoom level (25-200%)

nice catch I did not notice that

**Comment by p-carrier** [comment]
I fix the end date formatting plus add a null check with custom message for when it's null

Screenshot 2023-02-18 at 3 41 18 PM

**Pull request merged**